inginious/frontend/plugins/custom_input/static/custom_input.js

```diff
@@ -0,0 +1,76 @@
+function getTaskIdFromUrl() {
+    var urlTokens = window.location.pathname.split("/");
+    return urlTokens[urlTokens.length - 1];
+}
+
+function getCourseIdFromUrl() {
+    var urlTokens = window.location.pathname.split("/");
+    return urlTokens[urlTokens.length - 2];
+}
+
+function displayCustomTestAlertError(content) {
+    displayTaskStudentAlertWithProblems(content, "danger");
+}
+
+function runCustomTest (inputId) {
+    var customTestOutputArea = $('#customoutput-'+inputId);
+    var placeholderSpan = "<span class='placeholder-text'>Your output goes here</span>";
+
+    var runCustomTestCallBack = function (data) {
```

Comment:
Separate these functions because each one is doing different things, maybe you can move the `customTestOutputArea` variable as a global variable and in that way you'll have access to it on both functions.

Reply:
This function is a callback and we cannot modify its arguments. I would prefer to have the variable currified inside the function rather than a global